device/noise-protocol.go

```diff
@@ -97,15 +98,15 @@ type MessageCookieReply struct {
 type Handshake struct {
 	state                     int
 	mutex                     sync.RWMutex
-	hash                      [blake2s.Size]byte       // hash value
-	chainKey                  [blake2s.Size]byte       // chain key
-	presharedKey              NoiseSymmetricKey        // psk
-	localEphemeral            NoisePrivateKey          // ephemeral secret key
-	localIndex                uint32                   // used to clear hash-table
-	remoteIndex               uint32                   // index for sending
-	remoteStatic              NoisePublicKey           // long term key
-	remoteEphemeral           NoisePublicKey           // ephemeral public key
-	precomputedStaticStatic   [NoisePublicKeySize]byte // precomputed shared secret
+	hash                      [blake2s.Size]byte  // hash value
+	chainKey                  [blake2s.Size]byte  // chain key
+	presharedKey              wgcfg.SymmetricKey  // psk
+	localEphemeral            wgcfg.PrivateKey    // ephemeral secret key
+	localIndex                uint32              // used to clear hash-table
+	remoteIndex               uint32              // index for sending
+	remoteStatic              wgcfg.Key           // long term key
+	remoteEphemeral           wgcfg.Key           // ephemeral public key
```

Comment:
Going from 'PublicKey' to just 'Key' seems like a bit of a downgrade in clarity. Yes, in WireGuard, everything is 32-bytes, but the type is still important.

Reply:
Agreed. We still have the type distinction between private/public, but it would be nice to be a little clearer.

An experiment we ran in another repository for a different protocol was to create a `key` package with the types `Public` and `Private` in it. I'll experiment with that here and see if it looks good.

Reply:
Sounds complicated. Why not just have a PublicKey type and a PrivateKey type inside wgcfg?

In general, I don't think any wgcfg types belong in the Noise code. Cfg code is for configuration layer. Then Noise should deal with its own explicit types for crypto, like it was before. You can convert those types during assignment (they're all just byte arrays, anyhow), but intermingling it with wgcfg seems like bug potential.

Reply:
I explored how we've been using a `key` package and I agree it's more than is necessary here, so I'm going to rename `Key` back to `PublicKey`. Will do that in a followup commit.